Add Cloudflare Worker for serving tiles from PMTiles archives in R2

jake-low opened the pull request:
Adapted from Brandon's example code [here](https://github.com/protomaps/PMTiles/tree/main/serverless/cloudflare), with per-IP and per-Origin rate limiting added. I tested both rate limiting behaviors (manually lowering the rate limits so they were easier to trigger) and verified that they work.

Deployment instructions are in the README; basically `npm run deploy` to ship code changes.

quincylvania commented:
Cool! Thanks for working on this. I think the one thing missing is a whitelist of known origins that we don't rate limit (e.g. americanamap.org, openstreetmap.us). Also, what happens if the requester doesn't set an origin?

jake-low commented:
Good point re: a list of pre-approved origins; I'll work on adding that soon.

> Also, what happens if the requester doesn't set an origin?

Requests that don't have an `Origin` header will be rate-limited together in one bucket, since if the header is not present we use a fixed string (the empty string) as the rate limit key.

jake-low commented:
@quincylvania I've updated this branch to include functionality for favoring certain origins (i.e. not enforcing rate limits on them). The list of origins is stored in the Cloudflare dashboard rather than in the source code as a mild security precaution.

jake-low commented:
Added wildcard support to allowed and favored domain lists (so you can do things like `https://*.openstreetmap.us`) and switched to newlines instead of commas for delimiters. Latest version is deployed live if you want to test it (see osmus/OpenTrailMap#102 for example).

quincylvania commented:
@jake-low Awesome, thanks for making those updates!
